// Neon

syntax = "proto3";

package proto;

// --- Neon Interface

// Send a code update.
message PushCodeUpdateRequest {
  string code = 1;
}

message PushCodeUpdateResponse {}

// Send Midi events.
message PushMidiEventsRequest {
  bytes midi_payload = 1;
}

message PushMidiEventsResponse {}

// Get logs.
message GetLogsRequest {}

message GetLogsResponse {
  string source = 1;
  string notification = 2;
}

// Main Neon service.
service Neon {
  rpc PushMidiEvents(stream PushMidiEventsRequest) returns (PushMidiEventsResponse) {}
  rpc PushCodeUpdate(PushCodeUpdateRequest) returns (PushCodeUpdateResponse) {}
  rpc GetLogs(GetLogsRequest) returns (stream GetLogsResponse) {}
}
